### ISSUE TITLE
(🎁) `scripts_ctx` isn't optimal and needs two venvs

### ISSUE THREAD

**Issue by KotlinIsland** [pyprojectx/pyprojectx#130]
#126 was resolved with a solution of:

```toml
[tool.pyprojectx]
main = ["uv"]
project = ["-e ."]
scripts_dir = "scripts"
scripts_ctx = "project"
```

this means that there need to be two copies of the project and it's dependencies installed:
1. in `.venv` when you do `px uv sync`
2. in `.pyprojetx/venv/project-amogus0123456` when you do `px some-script`

this clearly isn't ideal, but i don't know what the best solution to this would be. ditch venv 1 and everything is just managed within a `project` context?

**Comment by houbie** [comment]
I'll give it another thought.
For the time being, there are 2 workarounds:
- define an alias
- use the scripts_ctx

**Comment by KotlinIsland** [comment]
> For the time being, there are 2 workarounds:
> 
> * define an alias
> * use the scripts_ctx

i'm not sure what you mean by this, what would the alias be? and the example is already using `scripts_ctx`

**Comment by houbie** [comment]
Actually it is already possible to reuse the project venv 😄 
Updated the [scripts](https://pyprojectx.github.io/config/scripts/) and [recipes](https://pyprojectx.github.io/recipes/#run-scripts-that-use-the-projects-packages) docs.

Feel free to reopen if not clear enough or if it doesn't work for you.

**Comment by KotlinIsland** [comment]
this is perfect, thank you

**Comment by KotlinIsland** [comment]
i'm not too sure these instructions are completely correct:
```toml
[tool.pyprojectx]
main = ["uv"]
scripts_dir = "scripts"
scripts_ctx = "venv"

[tool.pyprojectx.venv]
dir = ".venv"
post-install = "setup"

[tool.pyprojectx.aliases]
setup = "powershell scripts/setup"
```

```
> px example  # scripts/example.py
creating pyprojectx venv in C:\...
...
Successfully installed click-8.1.7 colorama-0.4.6 pyprojectx-3.1.2 tomlkit-0.13.2 userpath-1.9.2 uv-0.5.2
Using CPython 3.13.0 interpreter at: C:\...
Creating virtual environment at: .venv
Activate with: .venv\Scripts\activate
warning: Requirements file - does not contain any dependencies
Audited in 5ms
'setup' is not recognized as an internal or external command,
operable program or batch file.
PYPROJECTX ERROR: installation of 'venv' failed with exit code 1
```

**Comment by houbie** [comment]
I forgot to add an _@_ in the post-install do denote that it is a command to be ran with pw:
```toml
[tool.pyprojectx.venv]
dir = ".venv"
post-install = "@setup"
```
Does that work?

**Comment by houbie** [comment]
You need to indicate that _setup_ is an alias with an _@_:
```toml
[tool.pyprojectx.venv]
dir = ".venv"
post-install = "@setup"
```
I updated the docs accordingly.

This also revealed a bug when calling an alias in a post-install action, fixed in 3.1.3

Can you check again and close the issue if fixed?

**Comment by KotlinIsland** [comment]
that does work, thankyou
